table/block_based/block_type.h

```diff
@@ -14,10 +14,13 @@ namespace ROCKSDB_NAMESPACE {
 // Represents the types of blocks used in the block based table format.
 // See https://github.com/facebook/rocksdb/wiki/Rocksdb-BlockBasedTable-Format
 // for details.
-
+// For code sanity, BlockType should imply a specific TBlocklike for
+// BlocklikeTraits.
 enum class BlockType : uint8_t {
   kData,
-  kFilter,
+  kFilter,  // for second level partitioned filters and full filters
+  kFilterPartitionIndex,  // for top-level index of filter partitions
```

Comment:
Adding new types in between, would this be backward compatible or that is not needed.

Reply:
I don't think this enum value is persisted anywhere, and it's internal-only so binary compatibility with public API doesn't even come into play.